internal/handlers/handlers.go

```diff
@@ -1 +1,76 @@
 package handlers
+
+import (
+	"html/template"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/Yandex-Practicum/go1fl-sprint6-final/internal/service"
+)
+
+func IndexHandler(w http.ResponseWriter, r *http.Request) {
+	// Получаем текущую рабочую директорию
+	wd, err := os.Getwd()
+	if err != nil {
+		http.Error(w, "Failed to get working directory", http.StatusInternalServerError)
+		return
+	}
+	// Формируем абсолютный путь к index.html
+	indexPath := filepath.Join(wd, "index.html")
+	log.Printf("Looking for template at: %s", indexPath)
+
+	tmpl, err := template.ParseFiles(indexPath)
+	if err != nil {
+		log.Printf("Template error: %v", err)
+		http.Error(w, "Template not found", http.StatusInternalServerError)
+		return
+	}
+
+	if err := tmpl.Execute(w, nil); err != nil {
+		http.Error(w, "Template execution error", http.StatusInternalServerError)
+	}
+}
+
+func UploadHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodPost {
+		http.Redirect(w, r, "/", http.StatusSeeOther)
```

Comment:
Обычно если проблема с методом, то просто выставляют специальный код ошибки, редирект нужен когда пользователь что-то запрашивает (GET), а тут внутренний АПИ запрос и так делать не нужен
```suggestion
		w.WriteHeader(http.StatusMethodNotAllowed)
```

Reply:
Иван, привет!
Принято, спасибо большое!))